docker-app/qfieldcloud/filestorage/backend.py

```diff
@@ -1,5 +1,203 @@
+import base64
+import os
+
+import requests
+from django.core.files.base import ContentFile
+from django.core.files.storage import Storage
+from django.http import HttpResponse
 from storages.backends.s3boto3 import S3Boto3Storage
 
 
 class QfcS3Boto3Storage(S3Boto3Storage):
-    pass
+    def patch_nginx_download_redirect(self, response: HttpResponse) -> None:
+        pass
+
+
+class QfcWebDAVStorage(Storage):
+    """
+    Storage backend using WebDAV.
+    Adapted and inspired by this repository: https://github.com/marazmiki/django-webdav-storage
+    Copyright (c) 2020, Mikhail Porokhovnichenko
+    """
+
+    def __init__(self, **options):
+        self.requests = self.get_requests_session(**options)
+
+        self.webdav_url = options.get("webdav_url")
+        if not self.webdav_url:
+            raise NotImplementedError("Please define the webdav_url storage option")
```

Comment:
```suggestion
            raise NotImplementedError("Please define the `webdav_url` storage option")
```

Since it is now in our codebase, it should follow our approach of documenting stuff.

Also, I am not sure if `NotImplementedError` is the best choice? I guess we can check how the other storage providers do it?

Reply:
I guess `django-storage` is using `ImproperlyConfigured` exception type (https://github.com/jschneier/django-storages/blob/9de3cc9da4dbd67fcc56ecafc7cbf738af90136c/storages/base.py#L15C23-L15C43).

Thanks for spotting this, let's use it I guess.

Reply:
Thanks!